tooling/build/scripts/publisher.sh

```diff
@@ -37,21 +37,34 @@ echo $(git branch)
 # Perform a clean of npm cache
 npm cache clean --force
 
-# Install dependencies
-echo "Installing dependencies..."
-start_time=$(date +%s)
-npm ci
-calculate_duration $start_time
-
-# Build components
-echo "Building components..."
-start_time=$(date +%s)
-cd packages/components
-npm run build
-mv opengovsg-isomer-components-0.0.13.tgz ../../tooling/template/
-cd ../.. # back to root
-echo $(pwd)
-calculate_duration $start_time
+# Try to fetch cached node_modules from S3
+echo "Fetching cached node_modules..."
+NODE_MODULES_CACHE_PATH="s3://$S3_CACHE_BUCKET_NAME/$ISOMER_BUILD_REPO_BRANCH/isomer/node_modules.tar.gz"
+aws s3 cp $NODE_MODULES_CACHE_PATH node_modules.tar.gz || true
```

Comment:
issue: I think this will break our assumption that it takes at least 1 minute to reach the "fetching from database" step. Should we move that step further down after the "prebuilding" step and before the "building" step instead?

Reply:
@dcshzj thanks for catching this!

But moving it between prebuilding and building wouldn't help too since prebuilding is cached just like node_modules

With this PR, if cache is found (_which should be all builds except the first one after we publish a newer package version_), then the time to reach "fetching from database" right after the 1) queuing of build + 2) provisioning of instances. A safe min. time needed from observing past build is 20s (provisioning of instances)

Reply:
@dcshzj should we update it from 1 min to 30s?

Reply:
Ah oops I missed this comment, yeah if based on tests + moving the steps around it is still 30s then we should update that threshold too.